pkg/kubelet/cm/types.go

```diff
@@ -82,6 +82,14 @@ type CgroupManager interface {
 	ReduceCPULimits(cgroupName CgroupName) error
 	// MemoryUsage returns current memory usage of the specified cgroup, as read from the cgroupfs.
 	MemoryUsage(name CgroupName) (int64, error)
+	// GetCgroupMemoryConfig returns the memory limit of the specified cgroup as read from cgroup fs.
+	GetCgroupMemoryConfig(name CgroupName) (uint64, error)
```

Comment:
What's the benefit? The usage needs to query one thing at a time i.e memory limits or cpu limits/requests. If we changed it to generic, we'd have to read both CPU and memory whether or not both are needed. Why do extra work only to throw away the results?

Reply:
It is fine to have resource specific method.  I guess in concept we could have a generic method `GetResourceConfigFor(name CgroupName, []systems)` and return a sparse `ResourceConfig`, but I think most every call site would want a single resource at a time (cpu or memory) since pids, hugepages are not distinguished.  Maybe we do this update in the follow-on for v2 support as it can sparsely populate the `Unified` map.